### PR TITLE
Add a justfile

### PR DESCRIPTION
**Pull request by geoknee**
This centralises all scripts into one file with many benefits, including discoverability (try `just --list`). 

```
➜  superchain-registry git:(gk/just-so) ✗ just --list
Available recipes:
    add-chain                            # Adding a chain
    clean-add-chain                      # Clean test files generated by the add-chain tooling
    codegen                              # Generate auto-generated files
    lint-all                             # Lint all Go code
    promote-to-standard *args=''         # Promote a standard candidate chain to a standard chain, example: just promote-to-standard 10
    remove-chain SUPERCHAIN_TARGET CHAIN # Removing a chain, example: just remove-chain sepolia op
    test-add-chain                       # Test all Go code in the add-chain module
    test-all                             # Test all Go code
    test-superchain                      # Test all Go code in the superchain module
    test-validation                      # Test all Go code in the validation module
    tidy-add-chain                       # Tidy the add-chain go.mod file
    tidy-all                             # Tidying all go.mod files
    tidy-superchain                      # Tidy the superchain go.mod file
    tidy-validation                      # Tidy the validation go.mod file
    validate CHAIN                       # Run validation checks for chains with a name or chain ID matching the supplied regex, example: just validate 10
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,32 +18,25 @@ commands:
...
-      - run:  golangci-lint run superchain/... validation/... add-chain/...
-  golang-modules-tidy:
+      - run: |
+          wget -qO - 'https://proget.makedeb.org/debian-feeds/prebuilt-mpr.pub' | gpg --dearmor | sudo tee /usr/share/keyrings/prebuilt-mpr-archive-keyring.gpg 1> /dev/null
+          echo "deb [arch=all,$(dpkg --print-architecture) signed-by=/usr/share/keyrings/prebuilt-mpr-archive-keyring.gpg] https://proget.makedeb.org prebuilt-mpr $(lsb_release -cs)" | sudo tee /etc/apt/sources.list.d/prebuilt-mpr.list
+          sudo apt update
+          sudo apt install just
+
+jobs:
+  lint-and-tidy:
```
Comment (geoknee) [inline .circleci/config.yml:31]: We will need to sympathetically update the repo settings for "required" checks.

File: justfile [hunk 2]
```diff
@@ -0,0 +1,74 @@
+set dotenv-load
+set positional-arguments
+
+# Adding a chain
+add-chain:
+	go run ./add-chain
+	go run ./add-chain check-rollup-config
+	go run ./add-chain compress-genesis
+	go run ./add-chain check-genesis
```
Comment (bitwiseguy) [inline justfile:9]: This command has a required flag/env-var for the chain-id

File: README.md [hunk 3]
```diff
@@ -85,26 +80,25 @@ The format is a gzipped JSON `genesis.json` file, with either:
...
 or
 ```
-go test -run=TestValidation/11155420
+just validate 11155420
 ```
-Omit the `-run=` flag to run checks for all chains.
 
 > [!NOTE]
-> If you set `--standard-chain-candidate`, your chain will be checked against the majority of the standard configuration requirements. These are defined in the [specs](https://specs.optimism.io/protocol/configurability.html). However, these requirements are currently a draft, pending governance approval.
+> If you set `STANDARD_CHAIN_CANDIDATE`, your chain will be checked against the majority of the standard configuration requirements. These are defined in the [specs](https://specs.optimism.io/protocol/configurability.html). However, these requirements are currently a draft, pending governance approval.
```
Comment (bitwiseguy) [inline README.md:93]: The env vars are all prefixed with `SCR_` now. Suggest changing this note to:
```
If you set the env var SCR_ STANDARD_CHAIN_CANDIDATE=true...
```

File: justfile [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+# Test all Go code
+test-all: test-add-chain test-superchain test-validation
+
+# Test all Go code in the add-chain module
+test-add-chain:
+	# We separate the first test from the rest because it generates artifacts
+	# Which need to exist before the remaining tests run.
+	go test ./add-chain/... -run TestAddChain_Main -v
+	go test ./add-chain/... -run '[^TestAddChain_Main]' -v
+	just clean-add-chain
```
Comment (bitwiseguy) [inline justfile:34]: I think its nice to leave these test files around for sanity checking so I'd prefer to remove this line and just manually run `just clean-add-chain` when needed. Those files are already .gitignored so it shouldn't cause issues

File: justfile (hunk 2 above)
Comment (geoknee) [reply justfile:9]: [f21bee9](https://github.com/ethereum-optimism/superchain-registry/pull/366/commits/f21bee943b72f74b054e7597810972ca8685afde)

File: README.md (hunk 3 above)
Comment (geoknee) [reply README.md]: [a3eac59](https://github.com/ethereum-optimism/superchain-registry/pull/366/commits/a3eac59ce9d7c41be450719daec32768969263cc)

File: justfile (hunk 4 above)
Comment (geoknee) [reply justfile]: [e44827e](https://github.com/ethereum-optimism/superchain-registry/pull/366/commits/e44827e28025e0e6a7a50413d5b27ad25f2026be)

File: .circleci/config.yml (hunk 1 above)
Comment (geoknee) [reply .circleci/config.yml]: In the end I reverted to the existing pattern, we can change this later if we want to.

File: README.md [hunk 5]
```diff
@@ -31,34 +31,29 @@ This is to ensure your chain has a unique chain ID. Our validation suite will ch
...
-
-### 3. Run script
-
 #### Frontier chains
 
 Frontier chains are chains with customizations beyond the standard OP
 Stack configuration. To contribute a frontier OP-Stack chain
-configuration, you can run:
-
+configuration, you set the `CHAIN_TYPE=frontier` in the `.env` file.
```
Comment (geoknee) [inline README.md:45]:
```suggestion
configuration, you set the `SCR_CHAIN_TYPE=frontier` in the `.env` file.
```

File: README.md [hunk 6]
```diff
@@ -31,34 +31,29 @@ This is to ensure your chain has a unique chain ID. Our validation suite will ch
...
 
-```shell
-sh scripts/add-chain.sh frontier
-```
 
 #### Standard chains
 A chain may meet the definition of a **standard** chain. Adding a standard chain is a two-step process.
 
-First, the chain should be added as a frontier chain but with `--standard-chain-candidate` flag set:
+First, the chain should be added as a frontier chain as above, but with `STANDARD_CHAIN_CANDIDATE=true` in the `.env` file.
```
Comment (geoknee) [inline README.md:51]:
```suggestion
First, the chain should be added as a frontier chain as above, but with `SCR_STANDARD_CHAIN_CANDIDATE=true` in the `.env` file.
```

**Review by bitwiseguy — APPROVED** [review]
LGTM!

**Pull request merged**